roomserver/alias/alias.go

```diff
@@ -96,12 +96,14 @@ func (r *RoomserverAliasAPI) GetRoomIDForAlias(
 		return err
 	}
 
-	// No rooms found locally, try our application services by making a call to
-	// the appservice component
-	aliasReq := appserviceAPI.RoomAliasExistsRequest{Alias: request.Alias}
-	var aliasResp appserviceAPI.RoomAliasExistsResponse
-	if err = r.AppserviceAPI.RoomAliasExists(ctx, &aliasReq, &aliasResp); err != nil {
-		return err
+	if len(roomID) == 0 {
+		// No rooms found locally, try our application services by making a call to
+		// the appservice component
+		aliasReq := appserviceAPI.RoomAliasExistsRequest{Alias: request.Alias}
+		var aliasResp appserviceAPI.RoomAliasExistsResponse
+		if err = r.AppserviceAPI.RoomAliasExists(ctx, &aliasReq, &aliasResp); err != nil {
+			return err
+		}
 	}
 
 	response.RoomID = roomID
```

Comment:
If `aliasResp.RoomExists` is `true` we'll want to check again for whether the roomID exists in the db. That's because when we ask the application service whether the room exists, it can create one in the background before answering the request "yes, it exists! (now)".

So we'll need to check again after it does that. More info: https://matrix.org/docs/spec/application_service/unstable.html#get-matrix-app-v1-rooms-roomalias

Reply:
ok, cool. If `aliasResp.RoomExists` is `false`, should this method return an error instead of setting `response.RoomID` to an empty string?

Reply:
Nvm, just answered my own question by reading the method's callers